src/zenml/cli/stack.py

```diff
@@ -1250,3 +1278,316 @@ def register_secrets(
         cli_utils.declare(f"Updating secret `{secret.name}`:")
         cli_utils.pretty_print_secret(secret=secret, hide_secret=True)
         secrets_manager.update_secret(secret)
+
+
+@stack.command(help="Deploy a stack using mlstacks.")
+@click.option(
+    "--provider",
+    "-p",
+    "provider",
+    required=True,
+    type=click.Choice(STACK_RECIPE_MODULAR_RECIPES),
+)
+@click.option(
+    "--stack-name",
+    "-n",
+    "stack_name",
+    type=click.STRING,
+    required=True,
+    help="Set a name for the ZenML stack that will be imported from the YAML "
+    "configuration file which gets generated after deploying the stack recipe. "
+    "Defaults to the name of the stack recipe being deployed.",
+)
+@click.option(
+    "--region",
+    "-r",
+    "region",
+    type=click.STRING,
+    required=True,
```

Comment:
Does region make any sense for `k3d`?

Reply:
See above. We're locked into providing the region since `mlstacks` spec requires it. But for k3d, a local region would work just fine. In any case, the region isn't validated on our side.